src/components/TaskDetail.jsx

```diff
@@ -0,0 +1,299 @@
+        <div className="task-detail-header">
+          <h2>{task.content}</h2>
+          <div className="task-detail-actions">
+            {user && (
+              <button 
+                className={`delete-task-button ${showDeleteConfirm ? 'confirm' : ''}`}
+                onClick={handleDelete}
+              >
+                {showDeleteConfirm ? 'Click to confirm deletion' : 'Delete'}
+              </button>
+            )}
+            <button 
+              className="close-button" 
+              onClick={handleCloseClick}
+              title={showDeleteConfirm ? "Cancel deletion" : "Close"}
+            >
+              {showDeleteConfirm ? 'Cancel' : '×'}
+            </button>
+          </div>
+        </div>
+        
+        <div className="task-detail-content">
+          {/* Due Date Section */}
+          <div className="task-due-date-section">
+            <div className="due-date-header">
+              <h3>Due Date</h3>
+            </div>
+            {user && !showDeleteConfirm ? (
+              <DatePicker
+                selected={dueDate}
+                onChange={handleDueDateChange}
+                dateFormat="MMMM d, yyyy"
+                className="due-date-picker"
+                placeholderText="Set due date"
+                isClearable
+                showYearDropdown
+                scrollableYearDropdown
+              />
+            ) : (
+              <div className="due-date-display">
+                {dueDate ? new Date(dueDate).toLocaleDateString() : 'No due date set'}
+              </div>
+            )}
+          </div>
+
+          {/* Labels Section */}
+          <div className="task-labels-section">
+            <div className="labels-header">
+              <h3>Labels</h3>
+              {user && !showDeleteConfirm && (
+                <button 
+                  className="add-label-button"
+                  onClick={() => setShowLabelForm(!showLabelForm)}
+                >
+                  {showLabelForm ? 'Cancel' : 'Add Label'}
+                </button>
+              )}
+            </div>
+            
+            {showLabelForm && (
+              <div className="label-form">
+                <input
+                  type="text"
+                  value={newLabelText}
+                  onChange={(e) => setNewLabelText(e.target.value)}
+                  placeholder="Enter label text"
+                  className="label-input"
+                  maxLength={50}
+                  autoFocus
+                />
+                <div className="color-options">
+                  {LABEL_COLORS.map(color => (
+                    <button
+                      key={color}
+                      className={`color-option ${color === selectedColor ? 'selected' : ''}`}
+                      style={{ backgroundColor: color }}
+                      onClick={() => setSelectedColor(color)}
+                    />
+                  ))}
+                </div>
```

Comment:
![kody code-review](https://img.shields.io/badge/kody-code--review-brightgreen)

```suggestion
const ColorOptions = React.memo(({ selectedColor, onColorSelect }) => (
  LABEL_COLORS.map(color => (
    <button
      key={color}
      className={`color-option ${color === selectedColor ? 'selected' : ''}`}
      style={{ backgroundColor: color }}
      onClick={() => onColorSelect(color)}
    />
  ))
));

// Usage:
<ColorOptions
  selectedColor={selectedColor}
  onColorSelect={setSelectedColor}
/>
```
Consider memoizing the color options rendering to prevent unnecessary re-renders of the color buttons array.



<sub>Talk to Kody by mentioning @kody</sub>

<sub>Was this suggestion helpful? React with 👍 or 👎 to help Kody learn from this interaction.</sub><!-- kody-codereview -->&#8203;
&#8203;